### ISSUE TITLE
Dynamic widith needed when window title/name enabled

### ISSUE THREAD

**Issue by michaldybczak** [kotelnik/plasma-applet-active-window-control#69]
First, the widget version from manjaro repo 1.7.1-1 doesn't work well. Only after installing a version from AUR it started behaving rationally. In manjaro version, windows title/name was always behind buttons and many options didn't work. In AUR it's better but still many options don't do anything.

Widget width works great when only buttons are set. I made them appear with window maximized version. If a window is not maximized, buttons are gone and global menu shifts position correctly. When maximized, it shifts aside to make a space for buttons.

Enabling windows name/title won't overlapp with buttons in AUR version. But there is another problem: some applications have short names like spotify, dolphin, krusader, etc. But some have long ones, like Mail Client Thunderbird. This means, when I adjust position of buttons to long ones, I have bad looking empty space on other windows. When I set buttons for short names, long ones are just cut and that doesn't look good either.

I tried to enable 'fill widith' option but that screw everything because when it's on, global menu or other widgets on panel takes precedence and windows control is squeezed to a small corner with everything overlapping.

I'm not sure if the widget is not behaving as it should, maybe I set something wrong and maybe such behaviour is not implemented yet?

I would love to have full names/titles of windows and buttons next to it and dynamic width of a widget in corelation to global menu. However, without title/name option it's still good :).

Basically, there are two major things missing in the widget for full happiness ;)

1) lack of button look customization
2) no correct dynamic width when window name/title is enabled

**Comment by michaldybczak** [comment]
OK, I figured out something.

1) I needed to restart plasmashell to make a global menu from your widget work.
2) I needed to position buttons upper left. All other options make menu or window title overlapping
3) I can't use default global menu from plasma, only from your widget
4) I have to have AUR version of this widget
5) It's best to create a new panel after installation of this widget, on old panel something was off. I had set two panels, old and new one, and on new one, everything was fine, on old it was all messed.

If those conditions are met, it really works perfectly! Full dynamic and effects with window title/name. Just one option of the above set differently and I get an unholy mess...

It's hard to figure it out and it took me a lot of time. I bet most will give up after 10-15 min. I spend few hours playing around to finally get correctly working version of your widget with global menu and dynamic width.

I hope that my "adventures" and conclusions improve this widget or info on your site (how to install it, how to set it, what is the recommended version, ect.).

**Comment by MggMuggins** [comment]
Thanks for this issue, planning to configure it like this once I've finished installing the 10-28 update...
The version in the Manjaro Repos is now 1.7.3, so has the fixes you said you needed the AUR version for, so no need to install from AUR to get this functionality. Could you maybe share a screenshot of that part of your setup?
